### PR TITLE
erasure-code: fixes alignment issue when clay code is used with jerasure, cauchy_orig

### PR DESCRIPTION
**Pull request by mynaramana**
erasure-code:  fixes alignment issue when clay code is used with jerasure, cauchy_orig
doc: cleanup of clay code plugin documentation

Signed-off-by: Myna <mynaramana@gmail.com>

### CODE REVIEW

**Comment by mynaramana** [comment]
retest this please

File: doc/rados/operations/erasure-code-clay.rst [hunk 1]
```diff
@@ -1,8 +1,8 @@
-=========================
-Coupled LAYer code plugin
-=========================
+================
```
Comment (tchaikov) [inline doc/rados/operations/erasure-code-clay.rst]: i'd suggest you to move the doc changes in 864ccaee3231037dec2c6963949f3ad70a2ff719  into this commit . 

you could use `git reset HEAD^^` and `git add -p` to cherry-pick the change to add, for preparing the commits.

**Comment by tchaikov** [comment]
```
./non-regression.sh:94: run for func in $funcs
./non-regression.sh:95: run test_clay
./non-regression.sh:49: test_clay read k m d
./non-regression.sh:50: test_clay for stripe_width in $STRIPE_WIDTHS
./non-regression.sh:51: test_clay for technique in reed_sol_van
./non-regression.sh:52: test_clay non_regression --check --stripe-width 4096 --plugin clay --parameter scalar_mds=jerasure --parameter technique=reed_sol_van --parameter k=2 --pa
rameter m=2 --parameter d=3 --base .
./non-regression.sh:26: non_regression local action=--check
./non-regression.sh:27: non_regression shift
./non-regression.sh:29: non_regression test --check '!=' NOOP
./non-regression.sh:30: non_regression /var/ssd/ceph/build/bin/ceph_erasure_code_non_regression --check --stripe-width 4096 --plugin clay --parameter scalar_mds=jerasure --parame
ter technique=reed_sol_van --parameter k=2 --parameter m=2 --parameter d=3 --base .
2018-10-17 00:00:42.299 7f6cd039b640 -1 WARNING: all dangerous and experimental features are enabled.
2018-10-17 00:00:42.299 7f6cd039b640 -1 WARNING: all dangerous and experimental features are enabled.
2018-10-17 00:00:42.299 7f6cd039b640 -1 WARNING: all dangerous and experimental features are enabled.
2018-10-17 00:00:42.299 7f6cd039b640 -1 asok(0x55e407d5cf90) AdminSocketConfigObs::init: failed: AdminSocket::bind_and_listen: failed to bind the UNIX domain socket to '/tmp/user
/1000/ceph-asok.cKQeQP/client.admin.50738.asok': (2) No such file or directory
/var/ssd/ceph/src/erasure-code/clay/ErasureCodeClay.cc: In function 'int ErasureCodeClay::repair(const std::set<int>&, const std::map<int, ceph::buffer::list>&, std::map<int, cep
h::buffer::list>*, int)' thread 7f6cd039b640 time 2018-10-17 00:00:42.314606
/var/ssd/ceph/src/erasure-code/clay/ErasureCodeClay.cc: 421: FAILED ceph_assert(chunksize == (unsigned)chunk_size)
 ceph version Development (no_version) nautilus (dev)
 1: (ceph::__ceph_assert_fail(char const*, char const*, int, char const*)+0x1aa) [0x7f6cd4f70a28]
 2: (()+0x1221caa) [0x7f6cd4f70caa]
 3: (ErasureCodeClay::repair(std::set<int, std::less<int>, std::allocator<int> > const&, std::map<int, ceph::buffer::list, std::less<int>, std::allocator<std::pair<int const, cep
h::buffer::list> > > const&, std::map<int, ceph::buffer::list, std::less<int>, std::allocator<std::pair<int const, ceph::buffer::list> > >*, int)+0x16d) [0x7f6ccdee71e9]
 4: (ErasureCodeClay::decode(std::set<int, std::less<int>, std::allocator<int> > const&, std::map<int, ceph::buffer::list, std::less<int>, std::allocator<std::pair<int const, cep
h::buffer::list> > > const&, std::map<int, ceph::buffer::list, std::less<int>, std::allocator<std::pair<int const, ceph::buffer::list> > >*, int)+0x130) [0x7f6ccdee3c80]
 5: (ErasureCodeNonRegression::decode_erasures(std::shared_ptr<ceph::ErasureCodeInterface>, std::set<int, std::less<int>, std::allocator<int> >, std::map<int, ceph::buffer::list,
 std::less<int>, std::allocator<std::pair<int const, ceph::buffer::list> > >)+0x180) [0x55e4064c97d2]
 6: (ErasureCodeNonRegression::run_check()+0x5fa) [0x55e4064c9f6a]
 7: (ErasureCodeNonRegression::run()+0x61) [0x55e4064c8ec9]
 8: (main()+0x7b) [0x55e4064ca5cb]
 9: (__libc_start_main()+0xe7) [0x7f6cd30c4b17]
 10: (_start()+0x2a) [0x55e4064c7eda]
```

i still have the error above after applying this PR.

**Comment by mynaramana** [comment]
@tchaikov I am not able to see this issue. Could you try again ?

**Comment by mynaramana** [comment]
jenkins render docs

**Pull request merged**